File: samples/12_modular-kmp-ui-layer/iosApp/Sources/DetailsViewModel.swift
import FeatureTwoSharedUI

final class DetailsViewModel {
    private let repository: DetailsRepository

    init(repository: DetailsRepository = DetailsRepository(dataSource: DetailsDataSource())) {
        self.repository = repository
    }

    func load(platform: String) -> DetailsUiState {
        DetailsUiState(title: "Native DetailsViewModel on \(platform)", body: repository.loadDetailsSummary(platform: platform))
    }

    func refresh(platform: String) -> DetailsUiState {
        DetailsUiState(title: "Native DetailsViewModel refreshed on \(platform)", body: repository.loadDetailsSummary(platform: platform))
    }
}

File: samples/modular-kmp-ui-layer/androidApp/src/main/kotlin/com/example/kmpsamples/modularkmpuilayer/android/DetailsViewModel.kt
package com.example.kmpsamples.modularkmpuilayer.android

import com.example.kmpsamples.modularkmpuilayer.featuretwosharedui.DetailsUiState

class DetailsViewModel(
    private val repository: DetailsRepository = DetailsRepository(DetailsDataSource())
) {
    fun load(platform: String): DetailsUiState = DetailsUiState(
        title = "Native DetailsViewModel on $platform",
        body = repository.loadDetailsSummary(platform)
    )

    fun refresh(platform: String): DetailsUiState = DetailsUiState(
        title = "Native DetailsViewModel refreshed on $platform",
        body = repository.loadDetailsSummary(platform)
    )
}
